### PR TITLE
doc: Update readme

### PR DESCRIPTION
**Pull request by gaocegege**
Signed-off-by: Ce Gao <gaoce@caicloud.io>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/593)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,45 +40,91 @@ This project is inspired by [Google vizier](https://static.googleusercontent.com
 
 Katib stands for `secretary` in Arabic. As `Vizier` stands for a high official or a prime minister in Arabic, this project Katib is named in the honor of Vizier.
 
-## Concepts in Google Vizier
+## Concepts in Katib
 
-As in Google Vizier, Katib also has the concepts of Study, Trial and Suggestion.
+As in [Google Vizier][3], Katib also has the concepts of Experiment, Trial and Suggestion.
```
Comment (johnugeorge) [inline README.md]: Do we need to say Vizier in the doc? The terminologies have diverged in v1alpha2 and design has been changed

File: README.md [hunk 2]
```diff
@@ -7,20 +7,22 @@
 [![Coverage Status](https://coveralls.io/repos/github/kubeflow/katib/badge.svg?branch=master)](https://coveralls.io/github/kubeflow/katib?branch=master)
 [![Go Report Card](https://goreportcard.com/badge/github.com/kubeflow/katib)](https://goreportcard.com/report/github.com/kubeflow/katib)
 
-Hyperparameter Tuning on Kubernetes.
-This project is inspired by [Google vizier](https://static.googleusercontent.com/media/research.google.com/ja//pubs/archive/bcb15507f4b52991a0783013df4222240e942381.pdf). Katib is a scalable and flexible hyperparameter tuning framework and is tightly integrated with kubernetes. Also it does not depend on a specific Deep Learning framework (e.g. TensorFlow, MXNet, and PyTorch).
+Katib is a Kubernetes Native System for [Hyperparameter Tuning][1] and [Neural Architecture Search][2].
+The system is inspired by [Google vizier][3] and supports multiple ML/DL frameworks (e.g. TensorFlow, MXNet, and PyTorch).
```
Comment (johnugeorge) [inline README.md]: Same comment as below

File: README.md (hunk 1 above)
Comment (johnugeorge) [reply README.md]: With Suggestion CRD(next release), there is hardly any similarity with vizier

Comment (gaocegege) [reply README.md]: I think we could say it is inspired by vizier and do not introduce vizier in the doc, WDYT.

File: README.md [hunk 3]
```diff
@@ -38,45 +40,91 @@ This project is inspired by [Google vizier](https://static.googleusercontent.com
...
+
+Currently Katib supports the following exploration algorithms in v1alpha1:
 
-* random
-* grid
+* random search
+* grid search
 * [hyperband](https://arxiv.org/pdf/1603.06560.pdf)
 * [bayesian optimization](https://arxiv.org/pdf/1012.2599.pdf)
+* [NAS based on reinforcement learning](https://arxiv.org/abs/1611.01578)
```
Comment (hougangliu) [inline README.md]: In fact, our implementation doesn't follow exactly as the referred paper. wonder if we should note the gap

File: README.md (hunk 1 above)
Comment (johnugeorge) [reply README.md]: LGTM. I feel that we do not need to even mention 'vizier' when v1alpha1 is deprecated.

File: README.md (hunk 3 above)
Comment (gaocegege) [reply README.md]: SGTM

Comment (johnugeorge) [reply README.md]: https://github.com/kubeflow/katib/tree/master/pkg/suggestion/v1alpha1/NAS_Reinforcement_Learning

File: README.md (hunk 1 above)
Comment (gaocegege) [reply README.md]: I think our concepts about trial and study is inspired by vizier :thinking:

**Comment by gaocegege** [comment]
PTAL @johnugeorge @hougangliu 

Comments addressed.

File: README.md [hunk 4]
```diff
@@ -38,45 +40,90 @@ This project is inspired by [Google vizier](https://static.googleusercontent.com
...
-A Trial is a list of parameter values, x, that will lead to a single evaluation of f(x). A Trial can be “Completed”, which means that it has been evaluated and the objective value f(x) has been assigned to it, otherwise it is “Pending”.
-One trial corresponds to one job, and the job kind can be [k8s Job](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/), [TFJob](https://www.kubeflow.org/docs/guides/components/tftraining/) or [PyTorchJob](https://www.kubeflow.org/docs/guides/components/pytorch/), which depends on the Study's worker kind.
+A `Trial` is a list of parameter values, x, that will lead to a single evaluation of f(x). A Trial can be “Completed”, which means that it has been evaluated and the objective value f(x) has been assigned to it, otherwise it is “Pending”.
+
+In v1alpha1, `Trial` is just a concept inside Katib and not exposed to users.
+In v1alpha2, `Trial` is defined as a CRD `Trial` in Kubernetes.
+
+### Job
+
+`Job` is also known as `Worker` in [Google Vizier][3].
```
Comment (johnugeorge) [inline README.md]: Vizier reference?

**Comment by gaocegege** [comment]
Please take a look, thanks for your time :smile:

File: README.md [hunk 5]
```diff
@@ -38,45 +40,90 @@ This project is inspired by [Google vizier](https://static.googleusercontent.com
...
+
+Currently Katib supports the following exploration algorithms in v1alpha1:
 
-* random
-* grid
+* random search
+* grid search
 * [hyperband](https://arxiv.org/pdf/1603.06560.pdf)
 * [bayesian optimization](https://arxiv.org/pdf/1012.2599.pdf)
+* NAS based on reinforcement learning. The algorithm follows the idea proposed in [*Neural Architecture Search with Reinforcement Learning*](https://arxiv.org/abs/1611.01578), and the implementation is based on the open source implementation of [*Efﬁcient Neural Architecture Search via Parameter Sharing*](https://github.com/melodyguan/enas) (Also known as ENAS).
```
Comment (hougangliu) [inline README.md]: Sorry, I should have pointed just now more clear. Our nasrl misses some feature (saying layer skip) of enas.

Comment (gaocegege) [reply README.md]: Yeah, it is not ENAS, it is actually the original paper `Neural Architecture Search with Reinforcement Learning`. We noticed it here.

Comment (hougangliu) [reply README.md]: in our implementation the numLayers is fixed, the original paper has no this limitation

**Comment by gaocegege** [comment]
Comments are addressed, please take a look.
Thanks for your time :smile: @hougangliu

**Pull request merged**